src/overlays/canvas-overlay.js

```diff
@@ -18,22 +18,24 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-import {Component, createElement} from 'react';
+import {createElement} from 'react';
 import PropTypes from 'prop-types';
-import {PerspectiveMercatorViewport} from 'viewport-mercator-project';
+import BaseControl from '../components/base-control';
 import {window} from '../utils/globals';
 import autobind from '../utils/autobind';
 
-const propTypes = {
+const propTypes = Object.assign({}, BaseControl.propTypes, {
   redraw: PropTypes.func.isRequired
-};
+});
 
-const contextTypes = {
-  viewport: PropTypes.instanceOf(PerspectiveMercatorViewport),
-  isDragging: PropTypes.bool
+const defaultProps = {
```

Comment:
Why are these listed here instead of inherited?

Reply:
The overlays' default props are different from those of map controls.

Reply:
Main difference being the map controls (e.g. the NavigationControl and Popup) are expected to capture click and drag events. The overlays, on the other hand, are full screen. If any of these props are `true` the app will be broken by default.

Reply:
> The overlays' default props are different from those of map controls.

OK, I thought I had checked that they were the same, but you are right they are different. Thanks for clarifying.